packages/@glimmer/runtime/lib/vm/stack.ts

```diff
@@ -40,8 +40,8 @@ export class InnerStack {
   }
 
   write(pos: number, value: unknown): void {
-    if (isImmediate(value)) {
-      this.writeRaw(pos, encodeImmediate(value));
+    if (isImmediate(value) === true) {
```

Comment:
if isImmediate always returns a boolean then there is no coercion you are avoiding by doing this.